interfaces/apparmor/backend.go

```diff
@@ -426,6 +436,59 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	return errUnload
 }
 
+func (b *Backend) SetupMany(snaps []*snap.Info, confinement func(snapName string) interfaces.ConfinementOptions, repo *interfaces.Repository, tm timings.Measurer) []error {
+	var allChangedPaths, allUnchangedPaths, allRemovedPaths []string
+	var fallback bool
+	for _, snapInfo := range snaps {
+		opts := confinement(snapInfo.InstanceName())
+		changedPaths, removedPaths, unchangedPaths, err := b.prepareProfiles(snapInfo, opts, repo)
+		if err != nil {
+			fallback = true
+			break
+		}
+		allChangedPaths = append(allChangedPaths, changedPaths...)
+		allUnchangedPaths = append(allUnchangedPaths, unchangedPaths...)
+		allRemovedPaths = append(allRemovedPaths, removedPaths...)
+	}
+
+	if !fallback {
+		var errReloadChanged error
+		timings.Run(tm, "load-profiles-many[changed]", fmt.Sprintf("load changed security profiles of %d snaps", len(snaps)), func(nesttm timings.Measurer) {
+			errReloadChanged = loadProfiles(allChangedPaths, dirs.AppArmorCacheDir, skipReadCache|conserveCPU)
+		})
+
+		var errReloadOther error
+		timings.Run(tm, "load-profiles-many[unchanged]", fmt.Sprintf("load unchanged security profiles %d snaps", len(snaps)), func(nesttm timings.Measurer) {
+			errReloadOther = loadProfiles(allUnchangedPaths, dirs.AppArmorCacheDir, conserveCPU)
+		})
+		errUnload := unloadProfiles(allRemovedPaths, dirs.AppArmorCacheDir)
```

Comment:
is there a reason we don't also measure how long it takes to run `unloadProfiles`? is it because it is always very fast?

Reply:
> is there a reason we don't also measure how long it takes to run `unloadProfiles`? is it because it is always very fast?

Perhaps (idk why it isn't) but can say at this time unloadProfiles only consists of unlinking files, not unloading from the kernel (see interfaces/apparmor/apparmor.go:unloadProfiles() for details).

Reply:
Yes, that's exactly the reason, it's simply calling `UnlinkMany`, nothing to measure.